app/src/main/java/io/capibaras/abcall/data/repositories/IncidentsRepository.kt

```
@@ -0,0 +1,83 @@
+            } else {
+                val localData = incidentDAO.getAllIncidents()
+                if (localData.isNotEmpty()) {
+                    Result.success(localData)
+                } else {
+                    val errorBody = response.errorBody()!!.string()
+                    val jsonObject = JSONObject(errorBody)
+                    val message = jsonObject.getString("message")
+                    Result.failure(RepositoryError.CustomError(message))
+                }
+            }
+        } catch (e: IOException) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)